The goal is to apply descritpive and inferential statistics and hypothesis testing in python

Parts of notebook
Part 1: ETL
Part 2: Hypothesis Testing
Part 3: Results and Conclusion


# ETL



In [20]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

# Import packages for statistical analysis/hypothesis testing
from scipy import stats


In [ ]:
data = pd.read_csv("/Users/mycklandmatthew/Documents/Project Computer Science/Analytics 📊/2023_07_Coursera_Advanced Google Analytics/tiktok-truth-serum/data/tiktok_dataset.csv")


## Exploration

In [9]:
data.head()

# claim_status    video_id  video_duration_sec  \
0  1        claim  7017666017                  59   
1  2        claim  4014381136                  32   
2  3        claim  9859838091                  31   
3  4        claim  1866847991                  25   
4  5        claim  7105231098                  19   

                            video_transcription_text verified_status  \
0  someone shared with me that drone deliveries a...    not verified   
1  someone shared with me that there are more mic...    not verified   
2  someone shared with me that american industria...    not verified   
3  someone shared with me that the metro of st. p...    not verified   
4  someone shared with me that the number of busi...    not verified   

  author_ban_status  video_view_count  video_like_count  video_share_count  \
0      under review          343296.0           19425.0              241.0   
1            active          140877.0           77355.0            19034.0   
2            active          902185.0           97690.0             2858.0   
3            active          437506.0          239954.0            34812.0   
4            active           56167.0           34987.0             4110.0   

   video_download_count  video_comment_count  
0                   1.0                  0.0  
1                1161.0                684.0  
2                 833.0                329.0  
3                1234.0                584.0  
4                 547.0                152.0

In [10]:
data.describe()

#      video_id  video_duration_sec  video_view_count  \
count  19382.000000  1.938200e+04        19382.000000      19084.000000   
mean    9691.500000  5.627454e+09           32.421732     254708.558688   
std     5595.245794  2.536440e+09           16.229967     322893.280814   
min        1.000000  1.234959e+09            5.000000         20.000000   
25%     4846.250000  3.430417e+09           18.000000       4942.500000   
50%     9691.500000  5.618664e+09           32.000000       9954.500000   
75%    14536.750000  7.843960e+09           47.000000     504327.000000   
max    19382.000000  9.999873e+09           60.000000     999817.000000   

       video_like_count  video_share_count  video_download_count  \
count      19084.000000       19084.000000          19084.000000   
mean       84304.636030       16735.248323           1049.429627   
std       133420.546814       32036.174350           2004.299894   
min            0.000000           0.000000              0.000000   
25%          810.750000         115.000000              7.000000   
50%         3403.500000         717.000000             46.000000   
75%       125020.000000       18222.000000           1156.250000   
max       657830.000000      256130.000000          14994.000000   

       video_comment_count  
count         19084.000000  
mean            349.312146  
std             799.638865  
min               0.000000  
25%               1.000000  
50%               9.000000  
75%             292.000000  
max            9599.000000

In [11]:
# check missing values
data.isna().sum()

#                             0
claim_status                298
video_id                      0
video_duration_sec            0
video_transcription_text    298
verified_status               0
author_ban_status             0
video_view_count            298
video_like_count            298
video_share_count           298
video_download_count        298
video_comment_count         298
dtype: int64

Since there are only small numbers of missing values (298) compare to the number of all video 19382, the simplest way to deal is to trim and remove the data

In [14]:
# preserving the original data 
data_raw = data.copy()


In [15]:
#handiling the missing value by dropping
data = data.dropna()

In [17]:
# check missing values
data.isna().sum()

#                           0
claim_status                0
video_id                    0
video_duration_sec          0
video_transcription_text    0
verified_status             0
author_ban_status           0
video_view_count            0
video_like_count            0
video_share_count           0
video_download_count        0
video_comment_count         0
dtype: int64

The leadership team is interested in the relationship between verified_status and video_view_count. One approach is to examine the mean value of video_view_count for each group of verified_status in the sample data.

In [18]:
# Compute the mean `video_view_count` for each group in `verified_status`
data.groupby('verified_status')['video_view_count'].mean().reset_index(name='average_video_view_count')

verified_status  average_video_view_count
0    not verified             265663.785339
1        verified              91439.164167

# Hypothesis Testing


H0 = default hypothesis. There is no difference in video between verified versus unverified account

Ha = alternatives. There is a significance difference in video between verified versus unverified account

In [23]:
# determine the significance level of 95% 
alpha = 0.05

### Comparing two independent groups using two-sample t-test

In [21]:
# Two-sample t-test to compare means since we d
data_verified = data[data['verified_status']=='verified']
data_not_verified = data[data['verified_status']=='not verified']

tstat, pvalue = stats.ttest_ind(a=data_verified['video_view_count'], b=data_not_verified['video_view_count'], equal_var=False)

In [22]:
print(f't statistics: {tstat:f}\np-value: {pvalue:f}')

t statistics: -25.499442
p-value: 0.000000


In [25]:
# Determine the validity of the test
if pvalue < alpha:
    print(f'Since the p-value {pvalue:f} is less than the alpha {alpha}, we reject the null hypothesis.')
else:
    print(f'Since the p-value {pvalue:f} is greater than the alpha {alpha}, we fail to reject the null hypothesis.')

Since the p-value 0.000000 is less than the alpha 0.05, we reject the null hypothesis.


# Result and Conclusion

The hypothesis test shows that there is a statistically significant difference in the video view count between our two indpendent groups, the verified and unverified accounts.  This suggests that unverified accounts might be getting more views for specific reasons, which could be worth exploring further.

We want to build our model based on verified status. By analyzing the connection between verified status and other factors, we can gain insights into user behavior patterns that may help predict account claim status.